File: code/native/lib/repos/data/models/user/person.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:local/repos/data/models/location/location.dart';

part 'person.g.dart';

@JsonSerializable()
class Person {
  final int personID;
  final int userID;
  final DateTime createdAt;
  final String name;
  final String username;
  final String avatar;

  // generated
  final Location location;

  Person({
    required this.personID,
    required this.createdAt,
    required this.userID,
    required this.name,
    required this.avatar,
    required this.location,
    required this.username,
  });

  factory Person.fromJson(Map<String, dynamic> json) => _$PersonFromJson(json);

  Map<String, dynamic> toJson() => _$PersonToJson(this);
}

File: code/server/model/person.go
package model

import "time"

type Person struct {
	PersonID  int        `json:"personID,omitempty" db:"person_id"`
	UserID    int        `json:"userID,omitempty" db:"user_id"`
	Name      string     `json:"name,omitempty" db:"name"`
	Username  string     `json:"username,omitempty" db:"username"`
	Avatar    string     `json:"avatar,omitempty" db:"avatar"`
	Location  *Location  `json:"location,omitempty" db:"location"`
	CreatedAt *time.Time `json:"createdAt,omitempty" db:"created_at"`
}
